### PR TITLE
Move the stress test code to being a TestContainers-based test.

### PR DESCRIPTION
**Pull request by jkwatson**
Also: 

* Upgrade testcontainers version to work with the latest docker-for-mac.
* Add metrics in the OTLP exporter, to record success/failure of export.
* Use the TestContainers annotation to disable those tests when docker isn't present.

The only part I'm leery about here, is adding metrics to the exporter. I did find them super useful to gain insight into what the exporter thought had been exported, vs what the collector actually received, during network flakiness tests. I'd be happy to revert that piece for now, but I definitely think it could be a very useful addition that could help when debugging real-world issues.

### CODE REVIEW

File: integration_tests/src/test/java/io/opentelemetry/JaegerExporterIntegrationTest.java [hunk 1]
```diff
@@ -58,7 +59,7 @@
   @SuppressWarnings("rawtypes")
   @Container
   public static GenericContainer jaegerExampleAppContainer =
-      new GenericContainer("adoptopenjdk/openjdk8")
+      new GenericContainer(DockerImageName.parse("adoptopenjdk/openjdk8"))
```
Comment (anuraaga) [inline integration_tests/src/test/java/io/opentelemetry/JaegerExporterIntegrationTest.java:62]: Due to a bug in the RC of testcontainers, you need to add `:latest` here or notice the broken CI (it downloads every tag for the image ever right now...) Can you add `:lastest` to all the testcontainers untagged images?

Comment (jkwatson) [reply integration_tests/src/test/java/io/opentelemetry/JaegerExporterIntegrationTest.java:62]: oh blah. ok, I'll fix it

Comment (jkwatson) [reply integration_tests/src/test/java/io/opentelemetry/JaegerExporterIntegrationTest.java]: done. 🤞

**Review by anuraaga — APPROVED** [review]

**Comment by anuraaga** [comment]
EasyCLA seems to be dead :(

**Comment by jkwatson** [comment]
> EasyCLA seems to be dead :(

While we wait, any thoughts on adding the  metrics to the OTLP exporter?

**Comment by anuraaga** [comment]
Yeah the metrics seem important for exactly the reason you found :) I think I saw a spec issue about defining them but we could rename later as needed I think.

**Comment by jkwatson** [comment]
closing and reopening for the CLA bot

**Pull request merged**